src/main/solution_space/serialized_code.py

```diff
@@ -106,14 +106,19 @@ def get_anon_files(self, filter_anon_trees: Callable[[ast.AST], bool] = (lambda
     def get_canon_file(self) -> str:
         return self.__get_file_by_tree(self._canon_tree)
 
-    def __create_files_for_trees(self) -> None:
-        self.__create_file_for_tree(self._canon_tree, TREE_TYPE.CANON.value)
+    def rewrite_code_info(self, new_folder_with_files: str) -> None:
```

Comment:
Let's not mention 'code_info' here, because we have a class with the same name and it's a little bit unclear what do you mean by it there.
Also, you call the method 'create_files_for_tree', so it's clearer to use the same verb (recreate) instead of 'new' one (rewrite)
How about recreate_files_for_tree?